src/framework/components/layout-group/system.js

```diff
@@ -0,0 +1,136 @@
+
+    pc.extend(LayoutGroupComponentSystem.prototype, {
+        initializeComponentData: function (component, data, properties) {
+            if (data.orientation !== undefined) component.orientation = data.orientation;
+            if (data.reverseX !== undefined) component.reverseX = data.reverseX;
+            if (data.reverseY !== undefined) component.reverseY = data.reverseY;
+            if (data.alignment !== undefined) {
+                if (data.alignment instanceof pc.Vec2){
+                    component.alignment.copy(data.alignment);
+                } else {
+                    component.alignment.set(data.alignment[0], data.alignment[1]);
+                }
+                // force update
+                component.alignment = component.alignment;
+            }
+            if (data.padding !== undefined) {
+                if (data.padding instanceof pc.Vec4){
+                    component.padding.copy(data.padding);
+                } else {
+                    component.padding.set(data.padding[0], data.padding[1], data.padding[2], data.padding[3]);
+                }
+                // force update
+                component.padding = component.padding;
+            }
+            if (data.spacing !== undefined) {
+                if (data.spacing instanceof pc.Vec2){
+                    component.spacing.copy(data.spacing);
+                } else {
+                    component.spacing.set(data.spacing[0], data.spacing[1]);
+                }
+                // force update
+                component.spacing = component.spacing;
+            }
+            if (data.widthFitting !== undefined) component.widthFitting = data.widthFitting;
+            if (data.heightFitting !== undefined) component.heightFitting = data.heightFitting;
+            if (data.wrap !== undefined) component.wrap = data.wrap;
+
+            LayoutGroupComponentSystem._super.initializeComponentData.call(this, component, data, properties);
+
+            component.on('schedulereflow', this._onScheduleReflow, this);
+        },
+
+        cloneComponent: function (entity, clone) {
+            var layoutGroup = entity.layoutgroup;
+
+            return this.addComponent(clone, {
+                enabled: layoutGroup.enabled,
+                orientation: layoutGroup.orientation,
+                reverseX: layoutGroup.reverseX,
+                reverseY: layoutGroup.reverseY,
+                alignment: layoutGroup.alignment,
+                padding: layoutGroup.padding,
+                spacing: layoutGroup.spacing,
+                widthFitting: layoutGroup.widthFitting,
+                heightFitting: layoutGroup.heightFitting,
+                wrap: layoutGroup.wrap,
+            });
+        },
+
+        _onScheduleReflow: function (component) {
+            if (this._reflowQueue.indexOf(component) === -1) {
+                this._reflowQueue.push(component);
+            }
+        },
+
+        _onToolsUpdate: function () {
```

Comment:
Tools update is deprecated. I don't think this is ever called.

Reply:
It's definitely called – if I remove the `.on('toolsUpdate', ...` subscription then reflow never happens in the editor. Is there a new way I should be handling updates when in the editor?

Reply:
Hmm, so it is. Well, we still should avoid it. The calls regular update as well, so instead I think the better solution is to call `editor.call('viewport:render');` in the editor. This does a one frame update in the viewport.

Reply:
Hmm ok – where/when should that be called though? Whenever we change a relevant value in the editor?

Reply:
Ok – that does seem like a lot of extra code though; there are about 20 properties that would trigger a reflow (across `Element`, `LayoutGroup` and `Layout Child`). Also we'd have to keep things in sync – i.e. if a new property that should trigger a reflow is added/discovered, we'd need to remember to add it to the editor too.

Is there something about `toolsUpdate` that we didn't like that lead to the deprecation?

Reply:
There is a file in the Editor called viewport-entities-components-binding.js - you can do the viewport re-rendering there as that's where other components do the same too. We don't want to have Editor code in the Engine if possible I think that's why toolsUpdate shouldn't be used..